source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -199,6 +207,7 @@ class DecodingProcessorState : public ProcessorState {
       Filter& filter, const envoy::extensions::filters::http::ext_proc::v3::ProcessingMode& mode)
       : ProcessorState(filter) {
     setProcessingModeInternal(mode);
+    processor_type_ = ProcessorType::DecodingProcessor;
```

Comment:
Since the processor_type_ field belongs to the ProcessorState superclass, I think it would be cleaner to have it be an explicit parameter in the ProcessorState constructor, so then e.g. DecodingProcessorState could do `explicide DecodingProcessorState(...) : ProcessorState(filter, ProcessorType::DecodingProcessor) {}

This way there is structurally no way to have a ProcessorState instance whose processor_type_ field is unassigned.

Reply:
Is it ok to assign an UNSPECIFIED if we use the TrafficDirection enum?

Reply:
Either way works, so if you prefer that, SGTM.

My individual leaning would be to add the param to the ProcessorState constructor so that it can be set from the beginning to the proper value (and ProcessorState retains responsibility for setting its own fields--plus now you can make the field const), but that's subjective, without much real practical benefit, so up to you.